### PR TITLE
Add common flags for commands

### PR DESCRIPTION
**Pull request by wjhsf**
Added three common flags:
- `help`: the default help flag, but hidden, `-h` enabled, and description updated to have capital first letter
- `version`: the default version flag, but hidden, `-v` enabled, and description updated to have capital first letter
- `verbosity`: A flag that sets the log level, without requiring the command to have any special handling

First two are to appease @gkhunger-sf. Last one is one I made for #78. I ended up not including it in that PR because the logging situation is a bit complex. I think it's still worth using, though, at least until we address the disparate logging systems.

### CODE REVIEW

File: src/common/flags.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          ", "
+        )}`
+      );
+    }
+    return logger.getLevel();
+  }
+});
+
+/**
+ * The oclif help flag builder, with char set to -h, description changed to
```
Comment (unandyala) [inline src/common/flags.ts:42]:
```
description changed to
 * start with a capital letter,
```

Is this required in function doc?

File: src/common/flags.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    char: "h",
+    description: "Show CLI help",
+    hidden: true,
+    ...options
+  });
+}
+
+/**
+ * The oclif version flag builder, with char set to -v, description changed to
+ * start with a capital letter, and hidden by default.
```
Comment (unandyala) [inline src/common/flags.ts:59]: same here

File: src/common/flags.ts (hunk 1 above)
Comment (wjhsf) [reply src/common/flags.ts:42]: oclif uses lower case descriptions. We use sentence case. @gkhunger-sf didn't like the inconsistency, so this fixes it.

**Comment by dbarbato-sf** [comment]
As discussed offline with @wjhsf , let's drop the -v shortcut for version and assign -v to verbose, this makes the cli feel a bit more standard.

File: README.md [hunk 3]
```diff
@@ -45,17 +45,15 @@ ARGUMENTS
   NEW   The new API spec file (ruleset / diff-only mode) or configuration (directory mode)
 
 OPTIONS
-  -h, --help               show CLI help
-  -o, --out-file=out-file  File to store the computed difference
+  -V, --verbosity=trace|debug|info|warn|error|silent  [default: info] Set the level of detail in the output
```
Comment (dbarbato-sf) [inline README.md:48]: can we also make this --verbose not --verbosity ?

**Pull request merged**